subsys/net/lib/lwm2m/lwm2m_rw_json.c

```diff
@@ -605,8 +605,16 @@ int do_write_op_json(struct lwm2m_engine_obj *obj,
 
 			obj_field = lwm2m_get_engine_obj_field(obj,
 							       path->res_id);
+			/*
+			 * if obj_field is not found,
+			 * treat as an optional resource
+			 */
 			if (!obj_field) {
-				return -EINVAL;
+				if (context->operation != LWM2M_OP_CREATE) {
```

Comment:
I think both CREATE and WRITE operations should ignore the optional resource and continue on writing the following resource(s). If a server is writing multiple resources to the client w/ TLV or JSON format, this will return when the first optional resource is met and leave the rest unwritten.

Perhaps what we really want to do here is when any of the resource is written to the object instance (successfully) then we return w/ no error. Otherwise, we return w/ -ENOENT.